### PR TITLE
Support renaming views via alter table

### PR DESCRIPTION
**Pull request by mfussenegger**
Closes https://github.com/crate/crate/issues/14064

### CODE REVIEW

**Review by matriv — APPROVED** [review]
Nice!
Could we maybe as a followup support also the [ALTER VIEW x RENAME TO y](https://www.postgresql.org/docs/15/sql-alterview.html)?

**Comment by mfussenegger** [comment]
Added a second commit to change some formatting in the docs:

Before:
image

After:
image

File: docs/general/ddl/alter-table.rst [hunk 1]
```diff
@@ -210,12 +210,12 @@ reference documentation of :ref:`ALTER TABLE REROUTE
 
 Shard rerouting can help solve several problems:
 
-    * **Unassigned shards**: Due to cause of lack of space, shard awareness or
-      any other failure that happens during the automatic shard allocation it is
-      possible to gain unassigned shards in the cluster.
+* **Unassigned shards**: Due to cause of lack of space, shard awareness or
```
Comment (matriv) [inline docs/general/ddl/alter-table.rst:213]: :+1:

**Comment by mfussenegger** [comment]
retest this please

unrelated testKillUpdateByQuery failure

**Pull request merged**